### PR TITLE
Support querying olap table with StarOSTablet

### PR DESCRIPTION
**Pull request by wyb**
## What type of PR is this：
- [ ] bug
- [x] feature
- [ ] enhancement
- [ ] others

## Problem Summary(Required) ：

Select the all replicas in olap scan node for query plan.
The information of all replicas is obtained from StarOS.

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 3 / 17 (17.65%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/StarOSTablet.java|0|7|00.00%|
|:large_blue_circle:|com/starrocks/planner/OlapScanNode.java|2|8|25.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/Utils.java|1|2|50.00%|

**Comment by wyb** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 11 / 18 (61.11%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/planner/OlapScanNode.java|2|8|25.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/Utils.java|1|2|50.00%|
|:large_blue_circle:|com/starrocks/catalog/StarOSTablet.java|7|7|100.00%|
|:large_blue_circle:|com/starrocks/catalog/LocalTablet.java|1|1|100.00%|

File: fe/fe-core/src/main/java/com/starrocks/planner/OlapScanNode.java [hunk 1]
```diff
@@ -412,17 +414,21 @@ public void addScanRangeLocations(Partition partition,
             internalRange.setTablet_id(tabletId);
 
             // random shuffle List && only collect one copy
-            LocalTablet localTablet = (LocalTablet) tablet;
             List<Replica> allQueryableReplicas = Lists.newArrayList();
             List<Replica> localReplicas = Lists.newArrayList();
-            localTablet.getQueryableReplicas(allQueryableReplicas, localReplicas,
+            tablet.getQueryableReplicas(allQueryableReplicas, localReplicas,
```
Comment (wuleistarrocks) [inline fe/fe-core/src/main/java/com/starrocks/planner/OlapScanNode.java:419]: computeRangeLocations() also call getQueryableReplicas, maybe you should also fix it

Comment (wyb) [reply fe/fe-core/src/main/java/com/starrocks/planner/OlapScanNode.java:419]: Fixed

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 14 / 27 (51.85%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/planner/OlapScanNode.java|2|8|25.00%|
|:large_blue_circle:|com/starrocks/planner/MetaScanNode.java|3|9|33.33%|
|:large_blue_circle:|com/starrocks/sql/optimizer/Utils.java|1|2|50.00%|
|:large_blue_circle:|com/starrocks/catalog/StarOSTablet.java|7|7|100.00%|
|:large_blue_circle:|com/starrocks/catalog/LocalTablet.java|1|1|100.00%|

**Review by wuleistarrocks — APPROVED** [review]

**Review by gengjun-git — APPROVED** [review]

**Review by sduzh — APPROVED** [review]

**Pull request merged**